### PR TITLE
Add led block

### PR DESCRIPTION
**Pull request by aninternetof**
This adds a block to set the color of an RGB LED.

The code it outputs is `setChainableRgbLed(ledId, #FF2233)`. In the UI utility function, I'll parse this hex string into the three separate R, G, and B integers that the websocket protocol is detectin.g

There are tests for this repo. But, we're pretty far behind google/master, so I'll make a ticket to get us up-to-date, then we can figure out how the tests work then. I manually tested out this block in the playground file I committed.

### CODE REVIEW

**Comment by aninternetof** [comment]
The TravisJob you see was set up Google. In the future, it'd be great to get our stuff passing.

File: blocks/leds.js [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * @fileoverview Motor blocks for Blockly.
```
Comment (cabarnes) [inline blocks/leds.js]: LED

File: blocks/leds.js [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+Blockly.Blocks.leds.HUE = 77;
+
+Blockly.Blocks['chainable_rgb_led_set'] = {
+  init: function() {
+    this.appendDummyInput()
+        .appendField("set LED ");
+    this.appendValueInput("LED_ID")
+        .setCheck("Number");
+    this.appendDummyInput()
+        .appendField("to colour ");
```
Comment (cabarnes) [inline blocks/leds.js]: British `colour`?

File: generators/javascript/leds.js [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * @fileoverview Generating JavaScript for motor blocks.
```
Comment (cabarnes) [inline generators/javascript/leds.js]: LED

File: generators/javascript/leds.js [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+'use strict';
+
+goog.provide('Blockly.JavaScript.leds');
+
+goog.require('Blockly.JavaScript');
+
+Blockly.JavaScript['chainable_rgb_led_set'] = function(block) {
+  console.log(block);
+  var ledId = Blockly.JavaScript.valueToCode(block, 'LED_ID', Blockly.JavaScript.ORDER_ATOMIC);
+  var colour = Blockly.JavaScript.valueToCode(block, 'COLOUR', Blockly.JavaScript.ORDER_ATOMIC);
```
Comment (cabarnes) [inline generators/javascript/leds.js]: `COLOUR`

File: blocks/leds.js (hunk 1 above)
Comment (aninternetof) [reply blocks/leds.js]: fixing

File: blocks/leds.js (hunk 2 above)
Comment (aninternetof) [reply blocks/leds.js]: Ah, good catch. On the code side, the `setCheck` right below this expects `COLOUR`, so I figured I'd be consistent.
But this field is user-facing, so I'll change it to `color`.

File: generators/javascript/leds.js (hunk 4 above)
Comment (aninternetof) [reply generators/javascript/leds.js]: I'm gonna stick with the British on anything not user facing to be consistent with their code.

File: blocks/leds.js (hunk 2 above)
Comment (aninternetof) [reply blocks/leds.js]: Actually, the existing blocks that we'll use for picking colors are spelled `colour`, too, so I think I'd like to leave this.

File: generators/javascript/leds.js (hunk 3 above)
Comment (aninternetof) [reply generators/javascript/leds.js]: Fixed.

File: blocks/leds.js (hunk 2 above)
Comment (aninternetof) [reply blocks/leds.js]: image

**Pull request merged**